drainer/sync/kafka.go

```diff
@@ -32,23 +35,47 @@ var maxWaitTimeToSendMSG = time.Second * 30
 
 var _ Syncer = &KafkaSyncer{}
 
+// PartitionMode Kafka partition mode
+type PartitionMode byte
+
+const (
+	// PartitionFixed choose fixed partition 0
+	PartitionFixed PartitionMode = iota + 1
+	// PartitionBySchema choose partition by schema
+	PartitionBySchema
+	// PartitionByTable choose partition by table
+	PartitionByTable
+)
+
 // KafkaSyncer sync data to kafka
 type KafkaSyncer struct {
 	addr     []string
 	producer sarama.AsyncProducer
+	cli      sarama.Client
 	topic    string
 
-	toBeAckCommitTSMu sync.Mutex
-	toBeAckCommitTS   map[int64]struct{}
-
-	lastSuccessTime time.Time
+	msgTracker *msgTracker
 
 	shutdown chan struct{}
 	*baseSyncer
+
+	partitionMode     PartitionMode
+	includeResolvedTs bool
+}
+
+func partitionMode(mode string) PartitionMode {
+	switch strings.ToLower(mode) {
+	case "schema":
+		return PartitionBySchema
+	case "table":
+		return PartitionByTable
+	default:
```

Comment:
only use PartitionFixed if the mode is an empty string to avoid wrongly config like a typo?

Reply:
Fixed.